nomad/client_agent_endpoint.go

```diff
@@ -313,46 +314,18 @@ func (a *Agent) forwardFor(serverID, region string) (*serverParts, error) {
 }
 
 func (a *Agent) forwardMonitorClient(conn io.ReadWriteCloser, args cstructs.MonitorRequest, encoder *codec.Encoder, decoder *codec.Decoder) {
-	nodeID := args.NodeID
-
-	snap, err := a.srv.State().Snapshot()
-	if err != nil {
-		handleStreamResultError(err, nil, encoder)
-		return
-	}
+	// Get the Connection to the client either by fowarding to another server
+	// or creating direct stream
 
-	node, err := snap.NodeByID(nil, nodeID)
+	state, srv, err := a.findClientConn(args.NodeID)
```

Comment:
@drewbailey There was some common code here and in the pprof RPC endpoint and I needed it too. I want to draw your attention here in case I messed up your endpoints refactoring that common code out, especially in monitor because of the stream connection variation.